sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_client.hpp

```diff
@@ -45,17 +45,6 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
         std::shared_ptr<SharedKeyCredential> credential,
         const ShareClientOptions& options = ShareClientOptions());
 
-    /**
-     * @brief Initialize a new instance of ShareClient using token authentication.
-     * @param shareUri The URI of the file share this client's request targets.
-     * @param credential The client secret credential used to initialize the client.
-     * @param options Optional parameters used to initialize the client.
-     */
-    explicit ShareClient(
```

Comment:
Is there a header that needs to be removed as well?  or not?

Based on the changelog docs, we are removing `Azure::Identity::ClientSecretCredential` but this looks like a Core::TokenCredential.

It might be OK if TokenCredential is the base class for Azure::Identity::ClientSecretCredential. But then, shouldn't the changelog mention that we are removing all possible TokenCredentials?

Reply:
Agree. Line 8, (`#include "azure/identity/client_secret_credential.hpp"`)
` should be removed?

Regarding the changelog, IIRC the last time customers have seen this in public release, it was ClientSecretCredential, so it is fine to mention it as ClientSecretCredential then.